File: apps/core-api/src/Akademik.DataProvider/Models/Room.cs
using System.ComponentModel.DataAnnotations;

namespace Akademik.DataProvider.Models;

public sealed class Room
{
	public int Id { get; set; }
	public required string Number { get; set; }
	public int Floor { get; set; }
	[Range(0, 2)]
	public required int Capacity { get; set; }
	public RoomStatus Status { get; set; } = RoomStatus.Closed;
	public Assignment? Assignment { get; set; }
}

File: apps/web/src/shared/types/room.ts
export type RoomStatus = 0 | 1 | 2

export interface Room {
    id: number;
    number: string;
    floor: number;
    capacity: number;
    status: RoomStatus;
}

export interface RoomsListResponse {
    items: Room[];
    count: number;
}

export interface UpdateRoomRequest {
    id: number;
    number: string;
    floor: number;
    capacity: number;
    status: number;
}